streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -610,73 +610,79 @@ private StreamTask convertStandbyToActive(final StandbyTask standbyTask, final S
      * @throws StreamsException if the store's change log does not contain the partition
      * @return {@code true} if all tasks are fully restored
      */
-    boolean tryToCompleteRestoration(final long now, final java.util.function.Consumer<Set<TopicPartition>> offsetResetter) {
-        if (stateUpdater == null) {
-            boolean allRunning = true;
+    boolean tryToCompleteRestoration(final long now,
```

Comment:
Just to confirm there's no changes in `tryToCompleteRestoration` except the param right? The diffs are quite confusing... If it's the case I will skip comparing each line.

Reply:
I pulled `changelogReader.enforceRestoreActive()` and `changelogReader.transitToUpdateStandby()` from the stream thread into `tryToCompleteRestoration()` and removed the `if`-statement that separated the code path without the state updater from the code path with the state updater. I extracted the code path with the state updater into `checkStateUpdater()`.